test/instrumentation/modules/http/outgoing.js

```diff
@@ -230,6 +231,10 @@ function abortTest (type, handler) {
         t.equal(data.transactions.length, 1, 'has one transaction')
         t.equal(data.spans.length, 1, 'has one span')
         t.equal(data.spans[0].name, `${req.method} localhost:${port}/`, 'has expected span name')
+        if (typeof httpContext.status_code === 'undefined') {
```

Comment:
can you add else case which also checks for partial response? 

https://github.com/elastic/apm-agent-nodejs/blob/5bd3eb79fd806befb52c87bed1e1883229156918/test/instrumentation/modules/http/outgoing.js#L59-L71